ui/panels-plugin/src/plugins/gauge-chart/thresholds.ts

```diff
@@ -11,20 +11,73 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-export const ThresholdColorsMap = {
-  green: 'rgb(115, 191, 105)',
-  orange: 'rgba(237, 129, 40, 0.89)',
-  red: 'rgba(245, 54, 54, 0.9)',
+import { merge } from 'lodash-es';
+
+export const ThresholdColors = {
+  GREEN: 'rgba(115, 191, 105, 1)',
+  ORANGE: 'rgba(237, 129, 40, 0.9)',
+  RED: 'rgba(245, 54, 54, 0.9)',
 };
 
-export type ThresholdColorsType = keyof typeof ThresholdColorsMap;
+export type ThresholdColorsType = keyof typeof ThresholdColors | string;
 
 export type ThresholdOptions = {
   steps: StepOptions[];
+  default_color?: string;
 };
 
-type StepOptions = {
-  // color: string;
-  color: ThresholdColorsType;
+export type StepOptions = {
   value: number;
+  color: ThresholdColorsType;
 };
+
+export function convertThresholds(
+  thresholds: ThresholdOptions = {
+    steps: [{ value: 0, color: ThresholdColors.GREEN }],
+  }
+) {
+  const defaultThresholdColor = thresholds.default_color || ThresholdColors.GREEN;
+  const defaultThresholdArr: [number, string] = [0, defaultThresholdColor];
+  const defaultWarningColor = ThresholdColors.ORANGE;
+  const defaultAlertColor = ThresholdColors.RED;
+
+  const defaultThresholds: ThresholdOptions = {
+    default_color: defaultThresholdColor,
+    steps: [
+      {
+        value: 85,
+        color: defaultWarningColor,
+      },
+      {
+        value: 95,
+        color: defaultAlertColor,
+      },
+    ],
+  };
+
+  // converts to 2d array structure needed for series-gauge.axisLine.lineStyle.color
+  const stepsArr = thresholds.steps.map((step: StepOptions, index) => {
```

Comment:
don't you want to put this function outside of this function ? I'm thinking about unit test, and this function would be a good candidate. And  it would be cool to have some

Reply:
Unit test is added. Luke's zip approach suggestion reduced the number of lines in convertThresholds in half, so I left it as one function.